### ISSUE TITLE
clasp: No applicable method for REST-PARAMETER-BINDINGS with arguments

### ISSUE THREAD

**Issue by kpoeck** [robert-strandh/Concrete-Syntax-Tree#24]
```lisp
COMMON-LISP-USER> (macrolet ((%m (&rest (x y z)) `(quote (,x ,y ,z))))
    (%m 1 2 3))
; caught ERROR:
;   No applicable method for REST-PARAMETER-BINDINGS with arguments
;    (#<CLASP-CLEAVIR::CLASP-64BIT> #<DESTRUCTURING-LAMBDA-LIST children: (#<DESTRUCTURING-REQUIRED-PARAMETER-GROUP parameters: (#<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
;                                                                                                                                #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
;                                                                                                                                #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>) @0x12ea8d319>) @0x12ea94b19> #:G787534)

Condition of type: NO-APPLICABLE-METHOD-ERROR
No applicable method for REST-PARAMETER-BINDINGS with arguments
 (#<CLASP-CLEAVIR::CLASP-64BIT> #<DESTRUCTURING-LAMBDA-LIST children: (#<DESTRUCTURING-REQUIRED-PARAMETER-GROUP parameters: (#<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE> #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE> #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>) @0x12ea8d319>) @0x12ea94b19> #:G787534)
````

**Comment by kpoeck** [comment]
Fails here:
```lisp
(defmethod parameter-group-bindings
    (client (parameter-group destructuring-rest-parameter-group)
     argument-variable)
  (rest-parameter-bindings client (parameter parameter-group)
                           argument-variable))

(describe (parameter parameter-group)) -> 
<DESTRUCTURING-LAMBDA-LIST children: (#<DESTRUCTURING-REQUIRED-PARAMETER-GROUP parameters: (#<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
                                                                                             #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
                                                                                             #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>) @0x155f7d499>) @0x155f7d119> is an instance of class DESTRUCTURING-LAMBDA-LIST
it has the following instance slots
%CHILDREN:              (#<DESTRUCTURING-REQUIRED-PARAMETER-GROUP parameters: (#<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
                                                                               #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>
                                                                               #<CONCRETE-SYNTAX-TREE:SIMPLE-VARIABLE>) @0x155f7d499>)
; No value
````

**Comment by kpoeck** [comment]
I believe this is ost current code:
git log
commit 28dbbdd70dcc222062dadb156d8af305746cc1a8 (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Author: Jan Moringen <jmoringe@techfak.uni-bielefeld.de>
Date:   Sun Dec 15 15:29:47 2019 +0100

    RECONSTRUCT tracks previously seen CSTs correctly
    
    Before this change, ADD-ATOMS never entered anything into the SEEN
    table.
    
    fixes #21
